sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.getAssessmentNOA.e2e-spec.ts

```diff
@@ -0,0 +1,270 @@
+          disbursement1Status:
+            firstDisbursementSchedule.disbursementScheduleStatus,
+          disbursement1TuitionRemittance:
+            firstDisbursementSchedule.tuitionRemittanceRequestedAmount,
+          ...("disbursement1bcsl" in awards && {
+            disbursement1bcsl: awards.disbursement1bcsl,
+          }),
+          ...("disbursement1csgf" in awards && {
+            disbursement1csgf: awards.disbursement1csgf,
+          }),
+          ...("disbursement1cslf" in awards && {
+            disbursement1cslf: awards.disbursement1cslf,
+          }),
+        },
+        fullName: getUserFullName(application.student.user),
+        locationName: assessment.offering.institutionLocation.name,
+        noaApprovalStatus: assessment.noaApprovalStatus,
+        offeringIntensity: assessment.offering.offeringIntensity,
+        offeringStudyEndDate: getDateOnlyFormat(
+          assessment.offering.studyEndDate,
+        ),
+        offeringStudyStartDate: getDateOnlyFormat(
+          assessment.offering.studyStartDate,
+        ),
+        programName: assessment.offering.educationProgram.name,
+      });
+  });
+
+  it("Should throw unprocessable entity exception when assessment data is not found.", async () => {
+    // Arrange
+
+    // Student has an application to the institution eligible for NOA.
+    const student = await saveFakeStudent(db.dataSource);
+
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        state: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(currentMSFAA);
+    const application = await saveFakeApplicationDisbursements(db.dataSource, {
+      institution: collegeF,
+      institutionLocation: collegeFLocation,
+      student,
+      msfaaNumber: currentMSFAA,
+    });
+    application.currentAssessment.noaApprovalStatus = AssessmentStatus.required;
+    application.currentAssessment.assessmentData = null;
+    await db.application.save(application);
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/${application.currentAssessment.id}/noa`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        statusCode: 422,
+        message: "Notice of assessment data is not present.",
+        error: "Unprocessable Entity",
+      });
+  });
+
+  it("Should throw not found exception when assessment is not found.", async () => {
+    // Arrange
+
+    // Student has an application to the institution eligible for NOA.
+    const student = await saveFakeStudent(db.dataSource);
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        state: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(currentMSFAA);
```

Comment:
Do we need an MSFAA for this?

Reply:
https://github.com/bcgov/SIMS/pull/1952#discussion_r1207340054